lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -529,7 +530,10 @@ void ImageDecoderImpeller::Decode(fml::RefPtr<ImageDescriptor> descriptor,
         // Always decompress on the concurrent runner.
         auto bitmap_result = DecompressTexture(
             raw_descriptor, target_size, max_size_supported,
-            supports_wide_gamut, context->GetResourceAllocator());
+            /*supports_wide_gamut=*/supports_wide_gamut,
+            /*force_cpu_resize=*/
+            !context->GetCapabilities()->SupportsTextureToTextureBlits(),
```

Comment:
Can we get make this more apropos?  Can it just check if the backend is opengles?

Reply:
I think this is the right check. if we fix the texture blit issues then this lets us conditionally enable it for devices that support blit framebuffer

Reply:
`SupportsTextureToTextureBlits()` appears to be true only for the Metal backend.  We wouldn't want this for Vulkan, no?

Reply:
https://github.com/flutter/engine/blob/main/impeller/renderer/backend/vulkan/capabilities_vk.cc#L623-L625

Reply:
Okay, so this check is essentially a check for opengles.  Why is this a different check than `gl.BlitFramebuffer.IsAvailable()`?  Seems like they should be the same.

Reply:
Dependenices, we can't import gl.BlitFramebuffer here. But we also need to make it work first, and the capabilities lets me switch that off

Reply:
Right, I mean why wasn't CapabilitiesGLES like this:
```
bool CapabilitiesGLES::SupportsTextureToTextureBlits() const {
  return gl.BlitFramebuffer.IsAvailable();
}
```

Or a version check for version 3?  I think this code looks good considering everything, and since the Capabilities is just hardcoded to false we should be getting test coverage (assuming there is a golden test for this).  But if the reason `CapabilitiesGLES::SupportsTextureToTextureBlits()` is false is because of these weird issues we should explain it in a comment in `CapabilitiesGLES::SupportsTextureToTextureBlits()`.  I'm happy to write it, I'm just trying to understand why it was hardcoded to false before we identified these issues to make sure I understand everything.

Reply:
> Right, I mean why wasn't CapabilitiesGLES like this:

Because blit framebuffer is still misconfigured for the reisze path. Once that is fixed we can update the capabilities gles to do that.

Reply:
Done

Reply:
https://github.com/flutter/flutter/issues/158523